Assets/PC2D/Scripts/PlatformerMotor2D.cs

```diff
@@ -3321,5 +3275,155 @@ void OnDrawGizmosSelected()
         }
     }
 
+    private void ChangeState(MotorState newState) {

```

Comment:
This looks fine but good amount of testing to make sure delegates fire when expected.